AbletonMCP_Remote_Script/handlers/devices.py

```
@@ -0,0 +1,1360 @@
+    """Convert a sliced Simpler device to a Drum Rack."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
+        track = song.tracks[track_index]
+        if device_index < 0 or device_index >= len(track.devices):
+            raise IndexError("Device index out of range")
+        device = track.devices[device_index]
+        try:
+            from Live.Conversions import sliced_simpler_to_drum_rack as _convert
+        except ImportError:
+            raise Exception("sliced_simpler_to_drum_rack requires Live 12+")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)